### PR TITLE
Increase NNS_TENSOR_RANK_LIMIT @open sesame 04/06 10:37

### PR DESCRIPTION
**Pull request by niley7464**
This PR increases NNS_TENSOR_RANK_LIMIT to 8.

Tests to be modified and added

- [x] hardcoded dimensions (https://github.com/nnstreamer/nnstreamer/pull/3990)
- [x] unittest_common
- [x] tensor_converter
- [x] pytorch
- [x] tensorflow lite
- [x] transform_dimchg
- [x] transform_transpose
- [x] tensor_aggregator 
- [x] tizen_sensor
- [x] tensor_filter_lua (https://github.com/nnstreamer/nnstreamer/pull/3994)
- [x] cpp methods, filter extensions common, tizen_nnfw_runtime (https://github.com/nnstreamer/nnstreamer/pull/4000)
- [x] nnstreamer/api (https://github.com/nnstreamer/api/pull/241)
- [x] Tizen TCT
- [x] dimension in caps negotiation

If there are any tests that needed, please add it to the task list

See also: #3800 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Review by myungjoo — COMMENTED** [review]
I'd recommed to write an API at nnstreamer_plugin_api_util_impl.c (and a header too):

```gst_tensors_dimension_string_equal()``` for backward compatibility, which can compare dim_str.

**Review by anyj0527 — APPROVED** [review]
Great Thanks! LGTM ✔

File: gst/nnstreamer/nnstreamer_plugin_api_impl.c [hunk 1]
```diff
@@ -747,9 +762,13 @@ static GstCaps *
 _get_tensors_caps (const GstTensorsConfig * config)
 {
   GstCaps *caps;
+  GstStructure *structure;
 
   caps = gst_caps_from_string (GST_TENSORS_CAP_DEFAULT);
 
+  /* structure for backward compatibility */
```
Comment (songgot) [inline gst/nnstreamer/nnstreamer_plugin_api_impl.c:769]: How about adding a null check (config->info)?
dim_str(784 and 775) can be null if config->info is null.

**Comment by songgot** [comment]
Everything else seems like good code.

**Review by songgot — APPROVED** [review]

File: gst/nnstreamer/nnstreamer_plugin_api_impl.c (hunk 1 above)
Comment (niley7464) [reply gst/nnstreamer/nnstreamer_plugin_api_impl.c:775]: updated it
Thank you :)

**Review by again4you — APPROVED** [review]
Verified.
All tcts for Tizen are passed!

**Pull request merged**